### DISCUSSION TITLE
Co / Contravariant Interface Typeparameters - Pointer type

### DISCUSSION THREAD

**Discussion by bernd5** [dotnet/csharplang#6933] [Q&A]
Currently it is possible to define an interface like:

```csharp
unsafe interface IX<in T>
{
    void Do(T* data);
}
```

Is this really allowed?
I ask because we explicitly dissallow ref/in/out on non invariant parameter types...

**Comment by svick** [comment]
[It doesn't seem to be allowed by the runtime:](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmARgFgAoaosgTgAoMBPABxggDNGBJADQA8APgCUogNzVq+AM4BDLjAAEASzwYYULvLAqBg9coAqw6gG9qy68qIplAEQiNjAKmUATeRnmTqAXyA==)

> System.TypeLoadException: Could not load type 'IX`1' from assembly '...' because a covariant or contravariant type parameter was used illegally in the signature for an argument in method 'Do'.

The C# specification doesn't seem to be of much help here, since the variance rules are underspecified (see https://github.com/dotnet/csharpstandard/issues/195).
